games/hangman.c

```diff
@@ -0,0 +1,225 @@
+/**
+ * @file hangman.c
+ * @brief Hangman: The classic [Hangman](https://en.wikipedia.org/wiki/Hangman_(game)) word game
+ * @details Hangman is word guessing game where the program will randomly pick a word 
+ * from a word pool and represent it's letter in dashes"_ _ _". The user has to guess 
+ * the right word within a particular number of chances.
+ * @author [Anindya Sen](https://github.com/anindyasen)
+ */
+#include "hangman.h"
+
+
+/**
+ * @brief Generate a random index
+ *
+ * @param low Floor of index
+ * @param high Ceiling of index
+ * @return Randomly generated index
+ */
+int gen_index(int low, int high) { 
+    srand(time(0));
+    
+    return ((rand() % (high - low + 1)) + low); 
+} 
+
+/**
+ * @brief Display hangman game
+ *
+ * @param wg_cnt No. of wrong guess
+ * @param len Length of word
+ */
+void display(uint8_t wg_cnt, int len){
+    int i = 0;
+    printf("\033[H\033[J \n");
+    printf("GAME START\n");
+    printf(" ┏━━━┓\n");
+    printf(" ┃   │\n");
+    printf(" ┃   %c\n",   wg_cnt<1?' ':'O');
+    printf(" ┃  %c%s%c\n",wg_cnt<3?' ':'/', wg_cnt<2?" ":"│", wg_cnt<4?' ':'\\');
+    printf(" ┃  %c %c\n", wg_cnt<5?' ':'/',               wg_cnt<6?' ':'\\');
+    printf(" ┃\n");
+    printf("┏┻━━━━━━━┓\n┃   %d    ┗━┓\n┗━━━━━━━━━━┛\n",(WRONG_GUESS_CNT-wg_cnt));
+
+    for(i=0; i<len;i++) {
+        if(final_word[i] == 0){
+            printf("_ ");
+        }else {
+            printf("%c ",final_word[i]);
+        }
+    }
+    printf("\n");
+}
+
+/*
+ * Hash Table: Implemented as a array of linklist of size 21 (total no. of 
```

Comment:
Massive overcomplication; you don't need a hash table or a linked list for hangman at all! If you really need it, you should include it rather than duplicate it.